src/Compilers/VisualBasic/Portable/Symbols/ConstraintsHelper.vb

```diff
@@ -1086,28 +1091,55 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             Return New CompoundDiagnosticInfo(diagnostics)
         End Function
 
+        Public Enum ConstructorConstraintError
+            None
+            NoPublicParameterlessConstructor
+            MustInheritType
+            HasRequiredMembers
+        End Enum
+
         ''' <summary>
         ''' Return true if the class type has a public parameterless constructor.
         ''' </summary>
-        Public Function HasPublicParameterlessConstructor(type As NamedTypeSymbol) As Boolean
+        Public Function HasPublicParameterlessConstructor(type As NamedTypeSymbol, synthesizedIfMissing As Boolean) As ConstructorConstraintError
             type = type.OriginalDefinition
-            Debug.Assert(type.TypeKind = TypeKind.Class)
+            Debug.Assert(type.TypeKind = TypeKind.Class OrElse type.TypeKind = TypeKind.Structure)
 
             Dim sourceNamedType = TryCast(type, SourceNamedTypeSymbol)
 
-            If sourceNamedType IsNot Nothing AndAlso Not sourceNamedType.MembersHaveBeenCreated Then
+            If sourceNamedType IsNot Nothing AndAlso Not sourceNamedType.MembersHaveBeenCreated AndAlso sourceNamedType.TypeKind = TypeKind.Class Then
```

Comment:
>sourceNamedType.TypeKind = TypeKind.Class [](http://example.com/codeflow?start=104&length=41)

Consider checking this first and not even calculating ```sourceNamedType``` otherwise.